packages/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { createFilter } = require('@rollup/pluginutils');
+const { asyncWalk } = require('estree-walker');
+const MagicString = require('magic-string');
+
+function getRelativeAssetPath(node) {
+  return node.arguments[0].value;
+}
+
+// Looking for this:
+// new URL('./path/to/asset.ext', import.meta.url)
+function isNewUrlImportMetaUrl(node) {
+  return (
+    node.type === 'NewExpression' &&
+    node.callee.type === 'Identifier' &&
+    node.callee.name === 'URL' &&
+    node.arguments.length === 2 &&
+    node.arguments[0].type === 'Literal' &&
+    typeof getRelativeAssetPath(node) === 'string' &&
+    node.arguments[1].type === 'MemberExpression' &&
+    node.arguments[1].object.type === 'MetaProperty' &&
+    node.arguments[1].property.type === 'Identifier' &&
+    node.arguments[1].property.name === 'url'
+  );
+}
+
+function importMetaAssets({ include, exclude, warnOnError, transform } = {}) {
+  const filter = createFilter(include, exclude);
+
+  return {
+    name: 'rollup-plugin-import-meta-assets',
+
+    async transform(code, id) {
+      if (!filter(id)) {
+        return null;
+      }
+
+      const ast = this.parse(code);
+      const magicString = new MagicString(code);
+
+      await asyncWalk(ast, {
+        enter: async node => {
+          if (isNewUrlImportMetaUrl(node)) {
+            const { dir: absoluteScriptDir } = path.parse(id);
+            const relativeAssetPath = getRelativeAssetPath(node);
+            const absoluteAssetPath = path.resolve(absoluteScriptDir, relativeAssetPath);
+            const { base: assetName } = path.parse(absoluteAssetPath);
+
+            try {
+              const assetContents = await fs.promises.readFile(absoluteAssetPath);
```

Comment:
need to check if this works on node v10

Reply:
Will do. Official docs says v10. Local test with latest v10 works.